src/cmd_all/src/single_node.rs

```diff
@@ -134,30 +146,39 @@ pub fn map_single_node_opts_to_standalone_opts(opts: SingleNodeOpts) -> ParsedSt
         home_path.to_str().unwrap().to_string()
     });
 
-    // Set state store for meta (if not set)
+    // Set state store for meta (if not set). It could be set by environment variables before this.
     if meta_opts.state_store.is_none() {
-        let state_store_dir = format!("{}/state_store", &store_directory);
-        std::fs::create_dir_all(&state_store_dir).unwrap();
-        let state_store_url = format!("hummock+fs://{}", &state_store_dir);
-        meta_opts.state_store = Some(state_store_url);
+        if opts.in_memory {
+            meta_opts.state_store = Some("hummock+memory".to_string());
+        } else {
+            let state_store_dir = format!("{}/state_store", &store_directory);
+            std::fs::create_dir_all(&state_store_dir).unwrap();
+            let state_store_url = format!("hummock+fs://{}", &state_store_dir);
+            meta_opts.state_store = Some(state_store_url);
+        }
 
         // FIXME: otherwise it reports: missing system param "data_directory", but I think it should be set by this way...
         meta_opts.data_directory = Some("hummock_001".to_string());
     }
 
-    // Set meta store for meta (if not set)
+    // Set meta store for meta (if not set). It could be set by environment variables before this.
     let meta_backend_is_set = match meta_opts.backend {
         Some(MetaBackend::Etcd) => !meta_opts.etcd_endpoints.is_empty(),
         Some(MetaBackend::Sql) => meta_opts.sql_endpoint.is_some(),
         Some(MetaBackend::Mem) => true,
         None => false,
     };
     if !meta_backend_is_set {
-        meta_opts.backend = Some(MetaBackend::Sql);
-        let meta_store_dir = format!("{}/meta_store", &store_directory);
-        std::fs::create_dir_all(&meta_store_dir).unwrap();
-        let meta_store_endpoint = format!("sqlite://{}/single_node.db?mode=rwc", &meta_store_dir);
-        meta_opts.sql_endpoint = Some(meta_store_endpoint);
+        if opts.in_memory {
+            meta_opts.backend = Some(MetaBackend::Mem);
```

Comment:
What should we use, `Mem` (KV) or `Sql` with in-memory sqlite database?

Reply:
I just default to `Mem` since that's what playground uses, but I suppose in memory sqlite should be used in the future.